src/process.h

```diff
@@ -43,12 +43,12 @@ struct process {
     uint32_t entry;
     struct fd fd_table[PROCESS_MAX_OPEN_FILES];
     struct process *parent;
-    struct list children;
```

Comment:
There are several easy fixes: wrapping the child list objects in a struct with a pointer to the child, or using a process table and passing around PIDs instead of process pointers.